yt/_typing.py

```diff
@@ -1,5 +1,6 @@
 from typing import List, Optional, Tuple, Union
 
+import unyt as un
```

Comment:
You got me. The yt-wide convention is to always import from yt.units when possible, but I'm actively advocating against it, because it breaks typing: we're relying on indirect imports via `from unyt.<module> import *` statements in `yt.units`, which down the line forces us to add `# type: ignore` comments to imports from `yt.units`
I raised a similar point a couple years ago (before we started adding type hints) but was it rejected. I think maybe now is time to raise it again ?
Anyway in the context of the `yt._typing` module, it's a no brainer, importing from `yt.units` would simply not work for constraining types.